src/game/enemies/projectiles.c

```diff
@@ -0,0 +1,170 @@
+#include "global.h"
+#include "game/entity.h"
+#include "game/enemies/projectiles.h"
+#include "malloc_vram.h"
+#include "task.h"
+#include "trig.h"
+
+typedef struct {
+    /* 0x00 */ Sprite s;
+    /* 0x30 */ s32 x;
+    /* 0x34 */ s32 y;
+    /* 0x38 */ s16 velocityX;
+    /* 0x3A */ s16 velocityY;
+} ProjectileA; /* size: 0x3C */
+
+typedef struct {
+    /* 0x00 */ Sprite s;
+    /* 0x30 */ Vec2_32 positions[4];
+    /* 0x50 */ s16 velocities[4][2];
+    /* 0x60 */ u8 count;
+    /* 0x61 */ bool8 isActive[4];
+} ProjectileB; /* size: 0x68 */
+
+void Task_805102C(void);
+void Task_80510B0(void);
+void Task_DestroyProjectileTask(void);
+void TaskDestructor_80511EC(struct Task *);
+void TaskDestructor_8051200(struct Task *);
+
+void CreateProjectile(ProjInit *init)
+{
+    struct Task *t = TaskCreate(Task_805102C, sizeof(ProjectileA), 0x4000, 0,
+                                TaskDestructor_80511EC);
+    ProjectileA *proj = TaskGetStructPtr(t);
+    s32 velocityX, velocityY;
+
+    proj->x = init->x;
+    proj->y = init->y;
+
+    velocityX = (COS(init->rot) * init->speed) >> 14;
+    proj->velocityX = velocityX;
+
+    velocityY = (SIN(init->rot) * init->speed) >> 14;
+    proj->velocityY = velocityY;
+
+    proj->s.graphics.dest = VramMalloc(init->numTiles);
+    proj->s.graphics.anim = init->anim;
+    proj->s.variant = init->variant;
+    proj->s.unk1A = 0x200;
+    proj->s.graphics.size = 0;
+    proj->s.x = 0;
+    proj->s.y = 0;
+    proj->s.unk14 = 0;
+    proj->s.unk1C = 0;
+    proj->s.unk21 = 0xFF;
+    proj->s.unk22 = 0x10;
+    proj->s.palId = 0;
+    proj->s.unk28[0].unk0 = -1;
+    proj->s.unk10 = SPRITE_FLAG(PRIORITY, 1);
+}
+
+void CreateSeveralProjectiles(ProjInit *init, u8 count, s8 spreadAngle)
+{
+    struct Task *t = TaskCreate(Task_80510B0, sizeof(ProjectileB), 0x4000, 0,
+                                TaskDestructor_8051200);
+    ProjectileB *proj = TaskGetStructPtr(t);
+    u8 i;
+
+    if (count > 4)
+        count = 4;
+
+    proj->count = count;
+
+    for (i = 0; i < count; i++) {
+        s16 rot;
+        proj->positions[i].x = init->x;
+        proj->positions[i].y = init->y;
+
+        proj->velocities[i][0] = rot = ((i * spreadAngle) + init->rot) & ONE_CYCLE;
```

Comment:
Right.... this may help other matches! There might be a macro which assigns to s16